src/main/java/seedu/address/model/country/CountryNotesManager.java

```diff
@@ -72,6 +72,18 @@ public void deleteCountryNote(CountryNote countryNoteToDelete) {
         internalCountryNoteList.remove(countryNoteToDelete);
     }
 
+    /**
+     * Replaces the old country note with the new country note.
+     *
+     * @param oldCountryNote The old country note.
+     * @param newCountryNote The new country note.
+     */
+    public void setCountryNote(CountryNote oldCountryNote, CountryNote newCountryNote) {
+        int targetIndx = internalCountryNoteList.indexOf(oldCountryNote);
```

Comment:
I think indexOf can throw unchecked exception, so it would be better to do an assert contains before this line, and remove the assert after this line.

Reply:
Yup ok